config/ironic.conf.j2

```diff
@@ -1,20 +1,19 @@
 [DEFAULT]
 auth_strategy = noauth
 debug = true
-default_boot_interface = ipxe
```

Comment:
With this default removed, Ironic should pick the first supported boot interface in enabled_boot_interfaces below (according to https://docs.openstack.org/ironic/pike/install/enrollment.html#defaults-for-hardware-interfaces). Therefore,  removing this line and putting ipxe at the front of enabled_boot_interfaces is a noop for existing hardware, while allowing iLO to pick ilo-ipxe as a valid default boot interface.

Reply:
I don't think we necessarily need to remove the default, since this is set explicitly in the BMO?

https://github.com/metal3-io/baremetal-operator/blob/master/pkg/bmc/ilo5.go#L74
https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L411

Reply:
With 40495dc, I get an IncompatibleInterface exception before the Ironic API has started, and before BMO can interact with Ironic. The Ironic API container is instead stuck printing "Waiting for 9 expected drivers" (https://github.com/metal3-io/ironic-image/blob/216be3ee80c4dbc1a17bd94a9e6165f8f917836a/scripts/runironic-api#L13) and the conductor_hardware_interfaces table in MariaDB has no entries.

My guess is that the default value in the config is compared against all the hardware types when the Ironic conductor starts and is executing this method:

https://github.com/openstack/ironic/blob/4e931279865ed6abedc9a0837fcbf7234989d4d6/ironic/conductor/base_manager.py#L354

Reply:
This should be safe to remove since as @hardys correctly noticed we're setting the value explicitly (and iPXE is the default for most drivers anyway).